File: benchmarks/py_src/fib.py
#!/usr/bin/env python3
# Fibonacci benchmark — Python version
# Computes fib(N) using iterative approach
import sys

def fib(n):
    if n < 2:
        return n
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

if __name__ == "__main__":
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 35
    print(fib(n))

File: benchmarks/js_src/fib.js
// Fibonacci benchmark — Node.js (V8) version
// Computes fib(N) using iterative approach

function fib(n) {
    if (n < 2) return n;
    let a = 0, b = 1, c;
    for (let i = 2; i <= n; i++) {
        c = a + b;
        a = b;
        b = c;
    }
    return b;
}

const n = parseInt(process.argv[2] || '35', 10);
console.log(fib(n));
